net/suricata6/Makefile

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (C) 2006-2015 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=suricata
+PKG_VERSION:=6.0.0-beta1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_DATE:=2020-10-08
+PKG_SOURCE_VERSION:=95729e923f39b5adda31e4843a1b6ee06edbb0c8
+PKG_SOURCE_URL:=https://github.com/OISF/suricata.git
+PKG_HASH:=skip
+
+SURICATA-UPDATE_SOURCE_DATE:=2020-10-05
+SURICATA-UPDATE_SOURCE_VERSION:=86e530cbbfc78ac327072750a12edf888e6b8cae
+SURICATA-UPDATE_SOURCE_URL:=https://github.com/OISF/suricata-update.git
+SURICATA-UPDATE_HASH:=skip
+
+PKG_FIXUP:=autoreconf
+PKG_FIXUP:=patch-libtool
+
+PKG_INSTALL:=1
+HOST_BUILD_DEPENDS:=rust/host python3/host luajit/host
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/nls.mk
+include ../../lang/rust/rustc-triple.mk
+
+define Package/suricata6
+    SUBMENU:=Firewall
+    SECTION:=net
+    CATEGORY:=Network
+    DEPENDS:=+jansson +libcap-ng +libevent2 +libhiredis +liblz4 \
+      +libmagic +libpcap +libpcre +libmaxminddb +libnet-1.2.x \
+	 +libnetfilter-log +libnetfilter-queue +libnfnetlink +libnss \
+	 +libopenssl +luajit +python3 +python3-pip +python3-yaml \
+	 +python3-yaml-src +zlib $(ICONV_DEPENDS)
+    TITLE:=OISF Suricata IDS
+    URL:=https://www.openinfosecfoundation.org/
+endef
+
+CONFIGURE_VARS += \
+	CARGO_HOME=$(CARGO_HOME) \
+	ac_cv_path_CARGO="$(CARGO_HOME)/bin/cargo" \
+	ac_cv_path_RUSTC="$(CARGO_HOME)/bin/rustc"
+#	RUSTFLAGS="-C linker=$(TARGET_CC_NOCACHE) -C ar=$(TARGET_AR)"
+
+CONFIGURE_ARGS = \
+  	--prefix="/usr" \
+  	--sysconfdir="/etc" \
+  	--localstatedir="/var" \
+  	--enable-nfqueue \
+     --enable-af-packet \
+     --enable-unittests \
+	--enable-luajit \
+	--enable-geoip \
+     --enable-debug \
+	--enable-hiredis \
+     --enable-profiling \
+     --enable-profiling-locks \
+	--host=$(RUSTC_TARGET_ARCH) \
+	--build=$(RUSTC_HOST_ARCH) \
+	--with-suricata-update
+
+define Build/InstallDev
+	$(call HostPython3/Run,,$(HOST_PYTHON3_PIP_INSTALL_HOST) PyYAML)
+endef
+
+define Build/Prepare
+	$(call Build/Prepare/Default)
+	cd $(PKG_BUILD_DIR) && git clone https://github.com/OISF/libhtp.git
```

Comment:
Wouldn't it be better to package libhtp separately?
Something like this: https://gitlab.nic.cz/turris/turris-os-packages/-/blob/master/libs/libhtp/Makefile

Reply:
I thought about it, but then I've got to do ANOTHER PR just for LibHTP and wait for whoever to approve it.  I've already got a chain of contingencies here (Rust for Suricata for Suricata LibHTP and Update?) and didn't want to string them out to much.  I certainly don't MIND creating a libHTP - but since libHTP is used in the Suricata build directory on compile, I'll have to see about doing it that way if it's an issue :)